workspace/api/openfdd_bridge/routes/modbus_routes.py

```
@@ -82,3 +145,42 @@ async def modbus_read_registers(body: ModbusReadRequestBody) -> dict:
     except Exception as exc:
         raise HTTPException(status_code=502, detail=f"modbus_tcp_error: {exc}") from exc
+
+
+@router.post("/api/modbus/read_and_store", dependencies=[_READ])
+async def modbus_read_and_store(body: ModbusReadStoreBody) -> dict:
+    """Read Modbus TCP registers, append poll CSV, and write feather shard (source=modbus)."""
+    try:
+        payload = body.model_dump(exclude={"site_id", "save_registers"})
+        result = await asyncio.to_thread(execute_modbus_read_request, payload)
+        if body.save_registers:
+            for spec, reading in zip(body.registers, result.get("readings") or []):
```

Ruff (0.15.15)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)